datahub-web-react/src/app/search/SearchPage.tsx

```diff
@@ -69,6 +79,36 @@ export const SearchPage = () => {
         return refetch(variables).then((res) => res.data.searchAcrossEntities);
     };
 
+    const onChangeFilters = (newFilters: Array<FacetFilterInput>) => {
+        navigateToSearchUrl({ type: activeType, query, page: 1, filters: newFilters, history });
+    };
+
+    const onChangePage = (newPage: number) => {
+        navigateToSearchUrl({ type: activeType, query, page: newPage, filters, history });
+    };
+
+    /**
+     * Invoked when the "select all" checkbox is clicked.
+     *
+     * This method either adds the entire current page of search results to
+     * the list of selected entities, or removes the current page from the set of selected entities.
+     */
+    const onChangeSelectAll = (selected: boolean) => {
+        if (selected) {
+            // Add current page of urns to the master selected entity list
+            const entitiesToAdd = searchResultEntities.filter(
+                (entity) =>
+                    selectedEntities.findIndex(
+                        (element) => element.urn === entity.urn && element.type === entity.type,
+                    ) < 0,
+            );
+            setSelectedEntities(Array.from(new Set(selectedEntities.concat(entitiesToAdd))));
+        } else {
+            // Filter out the current page of entity urns from the list
+            setSelectedEntities(selectedEntities.filter((entity) => searchResultUrns.indexOf(entity.urn) === -1));
+        }
+    };
+
```

Comment:
so will checkboxes appear in the search results page itself? or is this tracking state for a modal that opens? If its tracking state for a modal, we should just have the modal track that state

Reply:
So yeah the check boxes appear in the search result list items... Then when you select an item we update the global state at this parent level so that the header check box and action list is also in sync!

Reply:
Not related to modal state- Can go deeper in person